File: hack1.rb
#ruby1.8.4@win32: 5.63, 97M(;4.98s, 69M)
#ruby1.8.5@cygwin: 4.77s, 93M(3.84s, 95M;3.16s, 68M)
#ruby1.8.5@ubuntu: 3.78s, 91M (3.36s, 91M;3.10s,64M)
#ruby1.8.5@mingw:4.807s,
#--------CD@3.34Ghz------------
#ruby1.8.7@ubuntu:3.10s,91M
#ruby1.9.1.20081030@ubuntu: 3.10s, 51M
#
def test
  count = Hash.new(0)
  File.read('test.txt').split.each {|word| count[word] += 1 }
  p count.sort_by{|_, x| x}.reverse.first(30)
end

if __FILE__ == $0
  t1 = Time.now
  test
  puts t2 = Time.now - t1
end

File: hack1.py
#@CD3.34Ghz Pc
#py2.5.2@xp: 2.04, 65M(1.51, 70M)
#py2.6@xp: 1.70, 65M(1.22,67M)
#irpy2.0@xp: 2.85, 188M
#py3.0rc2@xp: 14.95,89M
#--------------------------
#py2.5.2@ubuntu: 1.86,63M()
#py2.6@ubuntu: 1.90,63M(1.225)
#py3.0rc1+@ubuntu: 2.34,86M
#py3.0rc2@ubuntu: 2.39

from time import time
from operator import itemgetter
from heapq import nlargest
#import psyco
#psyco.full()

def test():
    histo = {} 
    for word in open("test.txt").read().split():
        histo[word] = 1 + histo.get(word, 0)
    print(nlargest(30, histo.items(), key=itemgetter(1)))

if __name__ == "__main__":
    t1 = time()
    test()
    print(time()-t1)
